apps/cowswap-frontend/src/modules/usdAmount/apis/getDefillamaUsdPrice.ts

```diff
@@ -0,0 +1,73 @@
+import { FractionUtils } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Fraction, Token } from '@uniswap/sdk-core'
+
+import ms from 'ms.macro'
+
+import { fetchWithRateLimit } from 'common/utils/fetch'
+
+import { RateLimitError, UnknownCurrencyError } from './errors'
+
+interface DefillamaUsdQuote {
+  coins: {
+    [chainAndAddress: string]: {
+      price: number
+    }
+  }
+}
+
+export const DEFILLAMA_PLATFORMS: Record<SupportedChainId, string | null> = {
+  [SupportedChainId.MAINNET]: 'ethereum',
+  [SupportedChainId.GNOSIS_CHAIN]: 'xdai',
+  [SupportedChainId.SEPOLIA]: null,
+}
+
+const BASE_URL = 'https://coins.llama.fi/prices/current'
+/**
+ * This is a text of 429 HTTP code
+ * https://saturncloud.io/blog/catching-javascript-fetch-failing-with-cloudflare-429-missing-cors-header/
+ */
+const FAILED_FETCH_ERROR = 'Failed to fetch'
+
+const fetchRateLimited = fetchWithRateLimit({
+  // Allow 2 requests per second
+  rateLimit: {
+    tokensPerInterval: 2,
+    interval: 'second',
+  },
+  // 2 retry attempts with 100ms delay
+  backoff: {
+    maxDelay: ms`0.1s`,
+    numOfAttempts: 2,
```

Comment:
is 2 attempts enough?

Reply:
Same logic as what's used on coingecko calls. If it's good there should be good here.